src/main/client/player/client-player-component.ts

```diff
@@ -20,45 +21,54 @@ export class ClientPlayerComponent {
 
    setClientPlayer(player: Player): void {
       this.clientId = player.id;
-      this.store.commit(player.id, player);
       this.clientInitSubject.next(player);
       this.store.update(player.id, player);
-      this.subscribePlayerSpriteCommitToStore();
       this.subscribeOnUpdateToPlayerSprite();
    }
 
    setClientPlayerSprite(player: PlayerSprite): void {
       this.clientPlayer = player;
    }
 
+   getClientId(): string | undefined {
+      return this.clientId;
```

Comment:
why would this clientId be `undefined`? should it not always be defined?

may lead to cant call X because its undefined errors later

Reply:
Must be but the variable is optional because it gets initialized only later.